### ISSUE TITLE
[Docs] Specify termguicolors option

### ISSUE THREAD

**Issue by j-steinbach** [famiu/feline.nvim#142]
Hi, I am a beginner nvim-user and I just spent three days trying to get feline to work. It just never showed colors, and I also never got any error messages. In the end I found out that I needed to do `o.termguicolors = true`... _(thanks for the minimal_init.lua!)_

The docs obviously list `Truecolor support for Neovim (do :help 'termguicolors' in Neovim for more info)` as a requirement, but it never ocured to me that this actually is a Neovim option. I thought it meant that I must run Neovim in a "true color terminal" -- which I do, which is why I never bothered reading into this deeper.

I propose clarifying the docs to list the specific color option.

- Truecolor support for Neovim (do :help 'termguicolors' in Neovim for more info). _Usually this involves setting `vim.o.termguicolors = true`._

Alternatively feline could check if that option is set on startup and throw an error.

----

For the curious: this is how it looks without colors working.

![2021-10-02_22-23-feline](https://user-images.githubusercontent.com/69524139/135899939-b836efa2-6185-4f28-bc82-22a5e4916e55.png)

**Comment by famiu** [comment]
I mean, isn't it the user's fault then for not using `:help 'termguicolors'` even though the docs clearly mentioned it? The reason I didn't show how to set the option directly is cause it can be done in 3 different ways (using `set termguicolors` or `vim.opt.termguicolors = true` or `vim.o.termguicolors = true`) and it depends on the user's preferences

**Comment by famiu** [comment]
Alright I implemented a check for `termguicolors`, closing this issue now

**Comment by j-steinbach** [comment]
Thank you for the fast response. I believe that a check is a good solution.

----

Regarding the docs: I think that your notion of `user fault` is technically correct, but I disagree with your idea of `clearly mentioned`. It is mentioned, but in an ambiguous way. There is no mention of having to `configure`something, only that I should inform myself. As I know that my terminal is capable of colors, and that my neovim also shows my terminal colors, I did not think that it was immediately necessary to do. _**I** found the instructions confusing._

TL:DR: The docs count 6000+ words. It should be clearly stated that there is an user action necessary.

----

An alternative to mentioning the above clearly in the "Requirements" section it could also be put into the "Help" section:

**No colors**

Did you set `termguicolors`? See `:h termguicolors` in vim or look at https://github.com/famiu/feline.nvim/issues/142

**Comment by famiu** [comment]
> Regarding the docs: I think that your notion of `user fault` is technically correct, but I disagree with your idea of `clearly mentioned`. It is mentioned, but in an ambiguous way. There is no mention of having to `configure`something, only that I should inform myself. As I know that my terminal is capable of colors, and that my neovim also shows my terminal colors, I did not think that it was immediately necessary to do. _**I** found the instructions confusing._
> 
> TL:DR: The docs count 6000+ words. It should be clearly stated that there is an user action necessary.

I suppose that's reasonable. I updated the README now, can you check it

**Comment by j-steinbach** [comment]
:+1:, sorry for being a bother. 

----

One last nit-pick: I think  "You must have 24-bit RGB color `configured` in Neovim" would be even clearer.

**Comment by famiu** [comment]
> One last nit-pick: I think "You must have 24-bit RGB color `configured` in Neovim" would be even clearer.

Configured to be what? Enabled? Disabled? Just adds more ambiguity imo

**Comment by j-steinbach** [comment]
You are right, it indeed is not clear. But my problem with just "enabled" is similar -- what does that mean? Where/how do I do that? `:h termguicolors` doesn't really say that this is to be done in the configuration either -- and I am speaking ultra-vim-noob here.

"You must have 24-bit RGB color enabled in your Neovim config".

**Comment by famiu** [comment]
> You are right, it indeed is not clear. But my problem with just "enabled" is similar -- what does that mean? Where/how do I do that? `:h termguicolors` doesn't really say that this is to be done in the configuration either -- and I am speaking ultra-vim-noob here.
> 
> "You must have 24-bit RGB color enabled in your Neovim config".

Again, the check for `termguicolors` already solves that issue, so I'm not going to change this again

**Comment by j-steinbach** [comment]
Fair enough. I simply continued replying as you asked me to comment on your change.
